src/Pages/Portofolio.jsx

```diff
@@ -0,0 +1,390 @@
+        quantity:
+          buttonName === "all" ? parseInt(quantity) : parseInt(amountInput),
+        type: "SELL",
+      };
+
+      console.log({assetToSell});
+
+      // TODO name for rutes: exchange/trade  rather than buy/sell
+
+      fetch(`${CALL_ME_SERVER}/trades/buy`, {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+          authorization: token,
+        },
+        body: JSON.stringify(assetToSell),
+      })
+        .then((res) => res.json())
+        .then((data) => {
+          console.log("SERVER POST SELL", data.asset);
+
+          // TODO:  create a function to keep track of the overall balance
+          // if (data.asset !== asset.quantity) {
+          //   const performance =
+          //     assetToSell.price * quantToSet - asset.price * quantToSet;
+          //   const balanceUpdated = balance + performance;
+          // }
+          window.alert(
+            "You have sold " + (buttonName === "all" ? quantity : amountInput) + " shares of " + assetSymbol
+          );
+        })
+        .catch((error) => console.error({ error }));
+    }
+  };
+
+  console.log({ portfolio });
+
+  const [order, setOrder] = React.useState("asc");
+  const [orderBy, setOrderBy] = React.useState("calories");
+  const [page, setPage] = React.useState(0);
+  const [dense, setDense] = React.useState(false);
+  const [rowsPerPage, setRowsPerPage] = React.useState(5);
+
+  const handleRequestSort = (event, property) => {
+    const isAsc = orderBy === property && order === "asc";
+    setOrder(isAsc ? "desc" : "asc");
+    setOrderBy(property);
+  };
+
+  const handleChangePage = (event, newPage) => {
+    setPage(newPage);
+  };
+
+  const handleChangeRowsPerPage = (event) => {
+    setRowsPerPage(parseInt(event.target.value, 10));
+    setPage(0);
+  };
+
+  const handleChangeDense = (event) => {
+    setDense(event.target.checked);
+  };
+
+  const emptyRows =
+    page > 0 ? Math.max(0, (1 + page) * rowsPerPage - portfolio.length) : 0;
+
+  return (
+    <Box className="table-container" sx={{ width: "100%" }}>
+      <Paper sx={{ width: "100%", mb: 2 }}>
+        {/* width of the actual form */}
+        <EnhancedTableToolbar /> {/* // here  Maybe to remove?  */}
+        <TableContainer>
+          <Table
+            sx={{ minWidth: 550 }}
+            aria-labelledby="tableTitle"
+            size={dense ? "small" : "medium"}
+          >
+            <EnhancedTableHead
+              order={order}
+              orderBy={orderBy}
+              onRequestSort={handleRequestSort}
+              rowCount={portfolio.length}
+            />
+            <TableBody>
+              {/* if you don't need to support IE11, you can replace the `stableSort` call with:
+                 rows.slice().sort(getComparator(order, orderBy)) */}
+              {stableSort(portfolio, getComparator(order, orderBy))
+                .slice(page * rowsPerPage, page * rowsPerPage + rowsPerPage)
```

Comment:
Some more copy-pasta, did you debug this to understand it?

Copying is fine as long as you do that

Reply:
copy 🍝
jk, I get line 296 as I had to make changes according to my needs so I get the `.slice()` but not really paid that much attention to line 295, the `getComparator` function